### PR TITLE
Don't omit schema metadata when removing column

### LINKED ISSUE #5327: Removing a column from a `RecordBatch` drops schema metadata
**Describe the bug**

Calling `remove_column` on a `RecordBatch` object deletes any schema metadata.

In https://github.com/apache/arrow-rs/blob/a0148ba4cea1756bd22c534dff14e362cb5ee242/arrow-array/src/record_batch.rs#L357-L362

It creates the new `SchemaBuilder` exclusively from the fields which calls this `From impl`

https://github.com/apache/arrow-rs/blob/a0148ba4cea1756bd22c534dff14e362cb5ee242/arrow-schema/src/schema.rs#L125-L132

instead of also taking into account the schema metadata.

**To Reproduce**

I don't have a full repro example but can create one if needed.

**Expected behavior**

Schema metadata not to be dropped

**Additional context**

### PR DESCRIPTION
**Pull request by kylebarron**
# Which issue does this PR close?

Closes #5327.

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by tustvold** [comment]
Could we get a test for this please

**Comment by kylebarron** [comment]
Apologies for the delay!

File: arrow-schema/src/schema.rs [hunk 1]
```diff
@@ -140,6 +140,15 @@ impl From<Fields> for SchemaBuilder {
     }
 }
 
+impl From<&Schema> for SchemaBuilder {
+    fn from(value: &Schema) -> Self {
+        Self {
+            fields: value.fields.to_vec(),
+            metadata: value.metadata.clone(),
+        }
```
Comment (alamb) [inline arrow-schema/src/schema.rs:148]: This could be slightly more concise 🤔 

```suggestion
        Self::from(value.clone())
```

**Review by alamb — APPROVED** [review]
Makes sense to me -- thank you @kylebarron

**Pull request merged**